cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -269,9 +275,9 @@ __global__ void computeSplitClassificationKernel(
   auto end = range_start + range_len;
   auto nclasses = input.nclasses;
   auto len = nbins * 2 * nclasses;
-  auto* shist = reinterpret_cast<int*>(smem);
-  auto* sbins = reinterpret_cast<DataT*>(shist + len);
-  auto* sDone = reinterpret_cast<int*>(sbins + nbins);
+  auto* shist = alignPointer<int*>(smem);
```

Comment:
this shouldn't be required as the start address of shared memory is typically always aligned.

Reply:
on a second thought. Please ignore my above comment. It is probably safer to keep this alignment here too.